api/src/db/db.service.ts

```diff
@@ -52,7 +60,12 @@ export class DbService {
                 reject("Invalid document: The passed document does not have an '_id' property");
             }
             this.getDoc(doc._id)
-                .then((existing: any) => {
+                .then((e) => {
+                    let existing;
+                    if (e.docs && e.docs.length > 0) {
+                        existing = e.docs[0];
+                    }
```

Comment:
Without the knowledge that this was done to avoid rewriting code this becomes a bit confusing. It would warrant a comment explanation I feel.

Reply:
Or rewrite the code of course 😄